src/NDItk/MultigroupTable/src/verify.hpp

```diff
@@ -8,14 +8,14 @@ void verify() {
 
   // consistent group structure
   const auto groups = this->metadata().numberGroups().value();
-  if ( ( this->structure().numberGroups() != groups ) ||
-       ( this->flux().numberGroups() != groups ) ||
+  if ( ( this->primaryGroupBoundaries().numberGroups() != groups ) ||
+       ( this->fluxWeights().numberGroups() != groups ) ||
        ( this->reactionCrossSections().numberGroups() != groups ) ) {
 
     Log::error( "Found inconsistent number of primary groups across the table" );
     Log::info( "Number of primary groups in the metadata: {}", this->metadata().numberGroups().value() );
-    Log::info( "Number of primary groups in the structure: {}", this->structure().numberGroups() );
-    Log::info( "Number of primary groups in the flux weights: {}", this->flux().numberGroups() );
+    Log::info( "Number of primary groups in the group structure: {}", this->primaryGroupBoundaries().numberGroups() );
+    Log::info( "Number of primary groups in the flux weights: {}", this->fluxWeights().numberGroups() );
     Log::info( "Number of primary groups in the reaction cross section data: {}",
```

Comment:
Should there be anything in here about outgoing particles, or does the fact that they're optional make them useless to verify?

Reply:
The outgoing particle type record is added later.